Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def resolve_url(url)
+    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
+    m.nil? ? url : resolve_url(m.captures.first)
```

Comment:
The issue is we can't know if and when such redirect would occur; it completely depends on the providers' actual configuration (f.e. as the they are gradually upgrading the internal elements of a redirect chain to SSL/TLS, or a secure mirror may be (mis)configured to redirect to a backup server without SSL/TLS), regardless of what the current configuration is. My vote for `--proto-redir`, to be on the secure side. But, even with `--proto-redir`, why manually parsing the headers of the first redirect is preferred over `-w '%{url_effective}'`?